middlewares/authorizeEventRoles.js

```diff
@@ -0,0 +1,50 @@
+const { ROLES } = require("../constants/events");
+const { getAllEventCodes } = require("../models/events");
+
+const userHasPermission = (userRoles, allowedRoles) => {
+  if (typeof userRoles === "object") {
+    return allowedRoles.some((role) => userRoles[`${role}`]);
+  } else if (typeof userRoles === "string") {
+    return allowedRoles.includes(userRoles);
+  }
+
+  return false;
+};
+
+const authorizeEventRoles = (allowedRoles) => {
+  return async (req, res, next) => {
+    const rolesAreValid = allowedRoles.every((role) => Object.values(ROLES).includes(role));
+    if (!rolesAreValid) {
+      return res.boom.badImplementation("Route authorization failed. Please contact admin");
+    }
+
+    if (req.userData) {
+      const roles = req.userData.roles;
+      if (roles) {
+        if (req.body.role === ROLES.HOST && userHasPermission(roles, allowedRoles)) {
+          return next();
+        } else if (req.body.role === ROLES.MODERATOR && userHasPermission(roles, allowedRoles) && roles.member) {
+          return next();
+        }
+      }
+    }
+
+    if (req.body.role === ROLES.MAVEN) {
+      const roomId = req.body.roomId ?? req.params.id;
+      const allowedEventCodes = await getAllEventCodes(roomId);
+      if (allowedEventCodes.includes(req.body.eventCode) && userHasPermission(req.body.role, allowedRoles)) {
+        return next();
+      }
+    }
```

Comment:
take latest pull and use the changes of mine that has been merged before for getEventCodes